conjure-java-core/src/main/java/com/palantir/conjure/java/util/Goethe.java

```diff
@@ -40,19 +44,43 @@ private Goethe() {}
 
     /** Formats the given Java file and emits it to the appropriate directory under {@code baseDir}. */
     public static Path formatAndEmit(JavaFile file, Path baseDir) {
+        StringBuilder code = new StringBuilder();
         try {
             Path outputFile = getFilePath(file, baseDir);
-            StringBuilder code = new StringBuilder();
             file.writeTo(code);
 
             CharSink sink = com.google.common.io.Files.asCharSink(outputFile.toFile(), StandardCharsets.UTF_8);
             JAVA_FORMATTER.formatSource(CharSource.wrap(code), sink);
             return outputFile;
-        } catch (IOException | FormatterException e) {
+        } catch (FormatterException e) {
+            throw new RuntimeException(generateMessage(file, code.toString(), e.diagnostics()), e);
```

Comment:
can we split this catch block to only apply to the JAVA_FORMATTER bit?
seems weird that this catch block relies on getting past `file.writeTo(code)` inside the same try block.

Reply:
ack, updated.